hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictorImpl.java

```diff
@@ -204,4 +221,208 @@ protected long getEvictionCriteriaValue(Record record, EvictionPolicy evictionPo
         return value;
     }
 
+
+    @Override
+    public boolean checkEvictionPossible(RecordStore recordStore) {
+        if (recordStore.size() < 1) {
+            return false;
+        }
+
+        String mapName = recordStore.getName();
+        int partitionId = recordStore.getPartitionId();
+
+        MapContainer mapContainer = recordStore.getMapContainer();
+        MaxSizeConfig maxSizeConfig = mapContainer.getMapConfig().getMaxSizeConfig();
+        MaxSizeConfig.MaxSizePolicy maxSizePolicy = maxSizeConfig.getMaxSizePolicy();
+
+        boolean result;
+        switch (maxSizePolicy) {
```

Comment:
I would not introduce a local variable with the maxSizePolicy. Just say

'''
switch(maxSizeConfig.getMaxSizePolicy()){
 ....
}
...

It reduces the amount of noise in the method. 

Another flavor things is to not store the result in the 'result' local variable, but immediately return from each case. Saves you another 2 lines on the method.